src/test/java/com/aws/greengrass/shadowmanager/sync/CloudDataClientTest.java

```diff
@@ -290,6 +290,7 @@ void GIVEN_100_synced_shadows_WHEN_unsubscribeForAllShadowsTopics_THEN_unsubscri
         }
 
         cloudDataClient.unsubscribeForAllShadowsTopics();
+        TimeUnit.MILLISECONDS.sleep(5000);
```

Comment:
what's this for?

Reply:
`cloudDataClient.unsubscribeForAllShadowsTopics();` now runs in a separate thread with the current changes.

Reply:
yeah ideally we have a way of waiting via a latch or something